blackboxopt/visualizations/visualizer.py

```diff
@@ -59,67 +125,42 @@ def multi_objective_visualization(evaluations: List[Evaluation]):
     if not evaluations:
         raise NoSuccessfulEvaluationsError
 
-    evaluation_dicts = [e.__dict__ for e in evaluations]
-
-    df = pd.DataFrame(evaluation_dicts)
-
-    # Used to structure the "sections" in the hover text, puts "info" section on top
-    hover_sections: Dict[str, list] = {
-        "info": [],
-        "configuration": [],
-        "optimizer_info": [],
-    }
-
-    # Expand and join json/dict columns
-    df = df.dropna(axis=0, subset=["objectives"]).reset_index(drop=True)
-    dfs_expanded = []
-    expand_column_names = ["configuration", "settings", "optimizer_info"]
-    for column in expand_column_names:
-        df_temp = pd.json_normalize(df[column], errors="ignore", max_level=0)
-        dfs_expanded.append(df_temp)
-        if column in hover_sections:
-            hover_sections[column] = sorted(df_temp.columns)
-    df = df.drop(expand_column_names, axis=1)
-    df = df.join(dfs_expanded, how="outer")
-
-    df_temp = pd.json_normalize(df["objectives"], errors="ignore", max_level=0)
-    objective_columns = list(df_temp.columns)
-    df = df.join(df_temp, how="outer")
-    df = df.drop("objectives", axis=1)
-
-    # Expand and join loss list column
-    df = df.dropna(axis=0, subset=objective_columns).reset_index(drop=True)
-
-    if len(df) <= 0:
-        raise NoSuccessfulEvaluationsError
-
-    # Calculate pareto info
-    df["pareto efficient"] = utils.mask_pareto_efficient(df[objective_columns].values)
+    # Prepare dataframe for visualization
+    df = evaluations_to_df(evaluations)
+    df["pareto efficient", "pareto efficient"] = utils.mask_pareto_efficient(
+        df["objectives"].values
+    )
 
-    # Cast datetimes (shown in hover text)
-    date_columns = ["finished_unixtime", "created_unixtime"]
-    df[date_columns] = df[date_columns].apply(pd.to_datetime, unit="s")
-    df["duration"] = df["finished_unixtime"] - df["created_unixtime"]
-    # Unclutter by rounding to seconds
+    # Formatting for nicer output
     df["duration"] = df["duration"].round("s").astype("str")
     df["created_unixtime"] = df["created_unixtime"].dt.round("1s")
```

Comment:
If we expect this to be necessary for most visualizations, we could also move this into the `evaluations_to_df()`. On the other hand it might be cumbersome if we use those columns as dimensions in some plots....